jenkinsfile-examples/cross-platform-build/Jenkinsfile

```diff
@@ -0,0 +1,174 @@
+#!groovy
+
+//
+// This DEMO shows how to write a cross-platform(Windows/Android/Linux/iOS/MacOSX) build.
+// Pipeline:
+//      1.Source code checkout -> 2.Cross-Platform Build -> 3.Archiving each platform's binaries -> 4.Cleanup
+//
+// We assume the project's structure is like this:
+//
+// --project_root
+//   --build
+//     -- android-build.sh (Script to build for Android platform, will output binaries into 'dist/android/armeabi-v7a')
+//     -- iphoneos-build.sh (Script to build for iOS, will output binaries into 'dist/ios')
+//     -- macos-build.sh (Script to build for MacOSX, will output binaries into 'dist/macos')
+//   --src
+//     -- CMakeLists.txt
+//     -- (Your source files structured by CMake or anyother)
+//   --dist
+//     -- (Directory to hold each platform distribution binaries.)
+//     -- android
+//     -- win32
+//     -- ios
+//     -- macos
+//   --symbols
+//     -- win32 (Dir to hold PDB symbols generate from CMake)
+//
+
+node('master') {
+
+    //
+    //  #1
+    //  Fresh checkout your cross-platform project onto the master node.
+    //  NOTE To avoid checkout on each platform again and again, we just checkout once on the master node,
+    //       then stash/unstash onto the target platform.
+    //
+    stage('Checkout') {
+        timeout(time: 10, unit: 'SECONDS') {
+            // Checkout source code onto master node.
+            // You can generate the 'checkout' command by using "Project->Pipeline Syntax->Snippet Generator"
+            checkout(GENERATE BY YOURSELF)
+            
+            // Stash all source files for another platform usage.
+            // NOTE: the stashed file's scope is in node 'master', so we can unstash it later.
+            stash name: 'YOUR_CROSSPLATFORM_SOURCES'
```

Comment:
For 'stash/unstash' topic:
In real uasge, I'm excluding much files than stashing whole worksapce, only source code was stashed, and Jenkins was setup on the same server which host my SVN, so checkout to a local workspace is fast enough for small and medium-sized project. And for this example, I removed 'exclude' session to simplify the syntax. Should we add 'exclude' back?

For 'checkout' topic:
This is a little complex, at the beginning, I'm following Jenkins guide to checkout(update clone after initial checkout) project on each platform, it works fine before I meet these two main problems:
1. Checkout/Update on each OS made me copy at least four times(Linux/Win32/MacOS/iOS) of the same 'checkout' code in the Jenkinsfile, it doesn't smells good to me.
2. According to the 1st reason, I'm wondering if there's any solution to avoid that copy, and I find that only 'stash/ubstash' works, but the drawback is the unstashed files is out of version control, so I need cleanup whole agent workspace after every build.

Currently the 'stash/unstash' solution works fine to me, but I'm still searching for a better one. Is there's any official examples or guidelines for cross-platform build pipleline template? Or I'll keep to make my branch better for general cross-platform usage. :)